.github/workflows/build.yaml

```
@@ -302,4 +310,62 @@ jobs:
       - name: Perform CodeQL Analysis
         uses: github/codeql-action/analyze@v2
         with:
-          category: "/language:java"
\ No newline at end of file
+          category: "/language:java"
+
+  deploy-snapshot:
+    name: Deploy Snapshot
+    if: ${{ github.ref == 'refs/heads/main' }}
+    needs: [codeql, test, pmd, spotbugs]
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Setup java
+        uses: actions/setup-java@v3
+        with:
+          distribution: "temurin"
+          java-version: "17"
+          server-id: artifactory-snapshots
+          server-username: ARTIFACTORY_USER # env variable for username in deploy
+          server-password: ARTIFACTORY_TOKEN # env variable for token in deploy
+      - name: Determine Maven Excludes
+        run: |
+          python .pipeline/scripts/get-maven-excludes.py --filter-key releaseAudience --filter-value None
+        id: get-maven-excludes-for-deploy-snapshot
+      # M2
+      - name: Load Maven Repository
+        uses: actions/cache/restore@v3
+        id: restore-cache
+        with:
+          path: ~/.m2/repository
+          key: maven
+      - name: Download M2 SDK Directory
+        uses: actions/download-artifact@v3
+        with:
+          name: m2-sdk
+          path: .
+      - name: UNZIP M2 SDK Directory
+        run: |
+          unzip m2-sdk.zip -d ~/.m2/repository/com/sap/cloud/sdk
+
+      # Target
+      - name: Download Target Dirs
+        uses: actions/download-artifact@v3
+        with:
+          name: target-dirs
+          path: .
+      - name: UNZIP Target Dirs
+        run: |
+          unzip -DD target.zip
```

[minor]

We could revert the change as it doesn't change anything I guess.
That way, the pipeline looks a bit more consistent.

```suggestion
          unzip -D target.zip
```

---

It at least avoids the recompile.